ui/packages/consul-ui/app/models/topology.js

```diff
@@ -33,14 +31,19 @@ export default class Topology extends Model {
     return undefinedDownstream;
   }
 
-  @computed('FilteredByACL', 'DefaultAllow', 'WildcardIntention', 'notDefinedIntention')
-  get collapsible() {
-    if (this.DefaultAllow && this.FilteredByACLs && this.notDefinedIntention) {
-      return true;
-    } else if (this.WildcardIntention && this.FilteredByACLs && this.notDefinedIntention) {
-      return true;
-    }
+  @computed('Downstreams', 'Upstreams')
+  // A service has a wildcard intention if `Allowed == true`  and `HasExact = false`
+  // The Permissive Intention notice appears if at least one upstream or downstream has
+  // a wildcard intention
+  get wildcardIntention() {
+    const downstreamWildcard =
+      this.Downstreams.filter(item => !item.Intention.HasExact && item.Intention.Allowed).length !==
+      0;
 
-    return false;
+    const upstreamWildcard =
+      this.Upstreams.filter(item => !item.Intention.HasExact && item.Intention.Allowed).length !==
+      0;
+
+    return downstreamWildcard || upstreamWildcard;
```

Comment:
Just a note here, not sure if it matters or not. Did you know we have all this intention information available to us in the API call we make for the Services > Intentions tab (we also use the same call to do the intention updating on this Topology Tab). I would guess reusing that would save a bunch of backend work here, and if the user was using TProxy I don't think (not entirely sure) we'd have to make the extra topology endpoint API call at all. Anyway, no action to be made on FEs part here, but if you are talking to BE about this it might free them up a little.